src/common/textentrycmn.cpp

```diff
@@ -229,8 +229,17 @@ void wxTextEntryBase::ChangeValue(const wxString& value)
 
 void wxTextEntryBase::AppendText(const wxString& text)
 {
+    bool old = false;
+    auto textCtrl = dynamic_cast<wxTextCtrl *>( this );
+    if( textCtrl && textCtrl->IsMultiLine() )
+    {
+        old = textCtrl->IsMaxLengthAllowed();
+        textCtrl->SetMaxLengthAllowed( false );
+    }
```

Comment:
Again, all this seems to be GTK-specific and has nothing to do in the common code.